tutorials/14-minituring/14_minituring.jmd

```diff
@@ -0,0 +1,327 @@
+---
+title: MiniTuring
+permalink: /:collection/:name/
+redirect_from: tutorials/14-minituring/
+weave_options:
+  error : false
+---
+
+In this session, we will develop a very simple probabilistic programming language.
+The implementation is similar to that of [Turing.jl's](https://turing.ml/dev/).
+This is intentional, as we want to demonstrate some key ideas from Turing's internal implementation.
+
+To make things easy to understand and to implement.
+We will restrict our language to a very simple subset of the language that Turing actually supports.
+Defining an accurate syntax description is not our goal here, instead, we will give a simple example, and all program that "looks like" this should work.
+
+For a model defined by the following mathematical form. (Consider `x` is data, or observed variable in this example.)
+
+$$
+\begin{align}
+a &\sim Normal(0.5, 1) \\
+b &\sim Normal(a, 2) \\
+x &\sim Normal(b, 0.5)
+\end{align}
+$$
+
+We will have the following model definition in our small language.
+
+```julia; eval=false
+@mini_model function m(x)
+    a ~ Normal(0.5, 1)
+    b ~ Normal(a, 2)
+    x ~ Normal(b, 0.5)
+end
+```
```

Comment:
Bluestyle requests a return value, but a return value here doesn't make sense. Is there a way to bypass the reviewdog on this?

Reply:
We could add a line with `#! format: off` above and `#! format: on` below the code. This could be hidden, I assume, by putting it into a `julia; echo=false` code block. Alternatively (and probably easier) would be to add a line `return nothing` to the model function.

Reply:
`julia; echo=false` will hide the code in the output; so just `return nothing` for now.